### PR TITLE
(RE-6121) Update to latest openssl

### PR DESCRIPTION
**Pull request by underscorgan**
Update to openssl 1.0.2e.

### CODE REVIEW

**Comment by underscorgan** [comment]
Test build (before rebase) http://jenkins-release.delivery.puppetlabs.net/job/vanagon_generic_job/239/

File: configs/platforms/solaris-10-i386.rb [hunk 1]
```diff
@@ -38,6 +38,8 @@
   pkgadd -n -a /var/tmp/vanagon-noask -G -d http://get.opencsw.org/now all;
   /opt/csw/bin/pkgutil -y -i curl rsync gmake pkgconfig ggrep;
   ln -sf /opt/csw/bin/rsync /usr/bin/rsync;
+  # RE-6121 openssl 1.0.2e requires functionality not in sytem grep
+  ln -sf /opt/csw/bin/ggrep /usr/bin/grep;
```
Comment (mckern) [inline configs/platforms/solaris-10-i386.rb]: Of course.

**Comment by mckern** [comment]
:+1:

File: configs/platforms/solaris-10-i386.rb (hunk 1 above)
Comment (haus) [inline configs/platforms/solaris-10-i386.rb]: What about aix? How is the grep there? Or does that work because /opt/freeware/bin is in PATH?

**Pull request merged**